features/ui/select/select.tsx

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @next/next/no-img-element */
+import React, { useState, MouseEventHandler, useEffect, useRef } from "react";
+import classNames from "classnames";
+import styles from "./select.module.scss";
+
+type Option = {
+  id: string;
+  value: string;
+};
+
+type SelectProps = {
+  optionsList: Option[];
+  logoSrc?: string;
+  label?: string;
+  hint?: string;
+  error?: Error;
+  disabled?: boolean;
+};
+
+export function Select({
+  optionsList,
+  logoSrc,
+  label,
+  error,
+  hint,
+  disabled,
+}: SelectProps) {
+  const [toggleSelect, setToggleSelect] = useState(false);
+  const [selectedValue, setSelectedValue] = useState<Option | null>(null);
+
+  const selectRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleMouseDown = (e: MouseEvent) => {
+      if (disabled) return;
+      if (selectRef.current && !selectRef.current.contains(e.target as Node)) {
+        setToggleSelect(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleMouseDown);
+
+    return () => document.removeEventListener("mousedown", handleMouseDown);
+  });
+
+  const handleToggleSelect: MouseEventHandler<HTMLButtonElement> = (e) => {
+    e.preventDefault();
+    if (disabled) return;
+    setToggleSelect(!toggleSelect);
+  };
+
+  const handleSelectValue = (option: Option) => {
+    if (disabled) return;
+    setSelectedValue(option);
+    setToggleSelect(false);
+  };
+
+  const renderOptions = () => {
+    return optionsList.map((option) => {
+      const isSelected = option.value === selectedValue?.value;
+      return (
+        <li key={option.id}>
+          <button
+            tabIndex={0}
+            className={classNames(isSelected && styles.isSelected)}
+            onClick={() => handleSelectValue(option)}
+          >
+            {option.value}
+            {isSelected && (
+              <img src="/icons/check-large.svg" alt="" aria-disabled />
+            )}
+          </button>
+        </li>
+      );
+    });
+  };
```

Comment:
I'm usually more a fan of creating a separate component instead of this kind of local render functions. Here you could create a separate Option component and keep the map inside the Select's JSX. But that might be a matter of personal taste